hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterQuota.java

```diff
@@ -1168,6 +1170,32 @@ public void testQuotaUpdateWhenDestinationNotPresent() throws Exception {
     assertEquals(0, quotaUsage.getSpaceConsumed());
   }
 
+  @Test
+  public void testAndByStorageType() {
+    long[] typeQuota = new long[StorageType.values().length];
+    Arrays.fill(typeQuota, HdfsConstants.QUOTA_DONT_SET);
+
+    Predicate<StorageType> predicate = new Predicate<StorageType>() {
+      @Override
+      public boolean test(StorageType storageType) {
+        return typeQuota[storageType.ordinal()] == HdfsConstants.QUOTA_DONT_SET;
+      }
+    };
+
+    assertTrue(Quota.andByStorageType(predicate));
+
+    typeQuota[0] = 2023;
+    assertFalse(Quota.andByStorageType(predicate));
+
+    Arrays.fill(typeQuota, HdfsConstants.QUOTA_DONT_SET);
+    typeQuota[1] = 2023;
+    assertFalse(Quota.andByStorageType(predicate));
+
+    Arrays.fill(typeQuota, HdfsConstants.QUOTA_DONT_SET);
+    typeQuota[typeQuota.length-1] = 2023;
+    assertFalse(Quota.andByStorageType(predicate));
```

Comment:
I recommend using a constant like HdfsConstants.QUOTA_RESET here instead of 2023.

Reply:
Ok, I've fixed it and added some necessary comments and I'm waiting for it to be recompiled, please review it later

![image](https://github.com/apache/hadoop/assets/65019264/a4de2978-c348-438c-84c7-af61e0e7d0f9)

Reply:
Thanks for the fix.